src/system.c

```diff
@@ -1519,8 +1512,50 @@ static void SySetGapRootPath( const Char * string )
         n++;
 #endif
     }
+    return;
 }
 
+/****************************************************************************
+**
+*F  SySetInitialGapRootPaths( <string> )  . . . . .  set the root directories
+**
+**  Setup GAP's initial root paths, based on two sources
+**  1) If the environment variable GAPROOT is undefined, the
+**     C macro GAPROOT_LOCATION (defined when GAP was compiled).
+**  2) Alternatively, try the location of GAP's executable.
+*/
+static void SySetInitialGapRootPaths(void)
+{
+    Int i, j;
+
+#ifdef GAPROOT_LOCATION
+    SySetGapRootPath(GAPROOT_LOCATION);
+    return;
+#endif
+
+    if (GAPExecLocation[0] != 0) {
+        /* GAPExecLocation might be a super-directory of GAP root,
+         * so we will go and search for the true GAP root.
+         * By default we try stepping back 2 levels */
+        char pathbuf[LONG_PATH_SIZE];
+        char initgbuf[LONG_PATH_SIZE];
+        for (i = 0; i < 3; ++i) {
+            strxcpy(pathbuf, GAPExecLocation, sizeof(pathbuf));
+            for (j = 0; j < i; ++j) {
+                strxcat(pathbuf, "../", sizeof(pathbuf));
+            }
+
+            strxcpy(initgbuf, pathbuf, sizeof(pathbuf));
+            strxcat(initgbuf, "lib/init.g", sizeof(pathbuf));
+
+            if (SyIsReadableFile(initgbuf) == 0) {
+                SySetGapRootPath(pathbuf);
+                /* escape from loop */
+                i = 4;
+            }
+        }
+    }
```

Comment:
Shouldn't there be a `SySetGapRootPath( "./" );` here (assuming we `return;` above)? Or at least a check to see if any root has been called, as it can happen that `SySetGapRootPath` never gets called now, can't it?